File: perl/types/strings/string_notation.pl
#!/usr/bin/perl

use strict;
use warnings;

my $s = "hello";
print("\"hello\": " . $s . "\n");

$s = 'hello';
print("'hello': " . $s . "\n");

$s = q\hello\;
print("q\\hello\\: " . $s . "\n");

$s = qq\hello\;
print("qq\\hello\\: " . $s . "\n");

$s = q|hello|;
print("q|hello|: " . $s . "\n");

$s = qq|hello|;
print("qq|hello|: " . $s . "\n");

File: php/strings/string_notation.php
#!/usr/bin/php8.0
<?php

$s = "hello";
print("\"hello\": " . $s . " (" . gettype($s) . ")\n");

$s = 'hello';
print("'hello': " . $s . " (" . gettype($s) . ")\n");
